jmix-flowui/flowui-kit/src/main/java/io/jmix/flowui/kit/component/grid/GridMenuItemActionWrapper.java

```diff
@@ -16,37 +16,74 @@
 
 package io.jmix.flowui.kit.component.grid;
 
-import com.vaadin.flow.component.*;
+import com.google.common.base.Strings;
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.Composite;
+import com.vaadin.flow.component.HasComponents;
+import com.vaadin.flow.component.HasText;
 import com.vaadin.flow.component.grid.contextmenu.GridContextMenu;
 import com.vaadin.flow.component.grid.contextmenu.GridMenuItem;
-import com.vaadin.flow.dom.Element;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.shared.HasPrefix;
+import com.vaadin.flow.component.shared.HasSuffix;
+import com.vaadin.flow.component.shared.HasTooltip;
+import com.vaadin.flow.component.shared.Tooltip;
 import com.vaadin.flow.shared.Registration;
 import io.jmix.flowui.kit.action.Action;
-
+import io.jmix.flowui.kit.component.HasAction;
 import jakarta.annotation.Nullable;
-import java.util.Optional;
 
-public class GridMenuItemActionWrapper<T> implements HasText, HasComponents, HasEnabled {
+public class GridMenuItemActionWrapper<T> extends Composite<Div>
+        implements HasText, HasComponents, HasPrefix, HasSuffix, HasTooltip, HasAction {
+
+    protected static final String ITEM_COMPONENT_CLASS_NAME = "jmix-grid-context-menu-item-component";
+    protected static final String PREFIX_COMPONENT_CLASS_NAME = "prefix-component";
+    protected static final String TEXT_COMPONENT_CLASS_NAME = "text-component";
+    protected static final String SUFFIX_COMPONENT_CLASS_NAME = "suffix-component";
 
-    protected final GridMenuItem<T> menuItem;
+    protected Span textComponent;
+    protected Component prefixComponent;
+    protected Component suffixComponent;
+    protected Tooltip tooltip;
+
+    protected GridMenuItem<T> menuItem;
 
     protected GridMenuItemActionSupport actionSupport;
 
+    public GridMenuItemActionWrapper() {
+    }
+
     public GridMenuItemActionWrapper(GridMenuItem<T> menuItem) {
         this.menuItem = menuItem;
     }
 
+    @Override
+    protected Div initContent() {
+        Div div = super.initContent();
+        div.setClassName(ITEM_COMPONENT_CLASS_NAME);
+
+        return div;
+    }
+
     public GridMenuItem<T> getMenuItem() {
         return menuItem;
     }
 
+    public void setMenuItem(GridMenuItem<T> menuItem) {
+        this.menuItem = menuItem;
+    }
+
     @Nullable
+    @Override
     public Action getAction() {
         return getActionSupport().getAction();
     }
 
-    public void setAction(@Nullable Action action) {
-        getActionSupport().setAction(action);
+    @Override
+    public void setAction(@Nullable Action action, boolean overrideComponentProperties) {
```

Comment:
you cann't remove prev public method. Revert `setAction(@Nullable Action action)` and call a new method from it with some default value for  `overrideComponentProperties`

Reply:
The class implements `HasAction` interface with default method just like you described:
```

public interface HasAction {

    default void setAction(@Nullable Action action) {
        setAction(action, true);
    }

    void setAction(@Nullable Action action, boolean overrideComponentProperties);

    @Nullable
    Action getAction();
}

```